imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -197,12 +290,61 @@ private void sendProcessCdrEvent(CallSummaryRecordDto record) {
     }
 
 
+    // Phase 1: verify the file exists, the csv is valid and its record count and checksum match the provided info
+    //          while collecting a list of errors on the go.
+    //          Does not proceed to phase 2 if any error occurred and returns an error
+    @Override
+    public VerifyResults verifyDetailFile(FileInfo fileInfo) {
+        File userDirectory = new File(System.getProperty("user.home"));
+        File cdrDirectory = new File(userDirectory, settingsFacade.getProperty(CDR_FILE_DIRECTORY));
```

Comment:
I'm pretty sure this won't work once we get to integration testing with the IVR.  They can't drop the file into the home directory of the server that also happens to be processing the http post.

We need to get clarity around the location of the file and how we will access that server.